oak-segment-tar/src/main/java/org/apache/jackrabbit/oak/segment/file/tar/TarWriter.java

```diff
@@ -183,7 +183,14 @@ void flush() throws IOException {
             }
 
             if (doFlush) {
-                archive.flush();
+                archive.flush(() -> {
+                    try {
+                        writeBinaryReferences();
```

Comment:
Do you know how much overhead this would introduce on each flush? Could this be problematic?

Reply:
I have run benchmark tests using Azure blob store (using "ConcurrentWriteTest" from oak-benchmark), and I haven't noticed degradation. 

Changed code:
```
(⎈|ethos13-prod-gbr9:ns-team-aem-cm-prd-n19203) oak-benchmarks - (OAK-10022) $ java -jar target/oak-benchmarks-*-SNAPSHOT.jar --azure "DefaultEndpointsProtocol=https;AccountName=smiljanisp;AccountKey=1bQTlsZ2JxjnT27g5AxamdGA0gXnVhNQWN7Xm8HrBGgv+T1taOj5a+dfqSK7ap9eGhtiAN+iTmorsKDrny1HYw==;" ConcurrentWriteTest  Oak-Segment-Azure 
Apache Jackrabbit Oak 1.45-SNAPSHOT
# ConcurrentWriteTest              C     min     10%     50%     90%     max     N       mean 
Oak-Segment-Azure                  1      53      60      77     494   1892     328     183
Oak-Segment-Azure                  1      45      60      73     477   1562     359     168
Oak-Segment-Azure                  1      53      63      76     388   1125     390     155
Oak-Segment-Azure                  1      53      64     102     521   1594     285     211
```
Trunk:
```
(⎈|ethos13-prod-gbr9:ns-team-aem-cm-prd-n19203) oak-benchmarks - (trunk) $ java -jar target/oak-benchmarks-*-SNAPSHOT.jar benchmark  --azure "DefaultEndpointsProtocol=https;AccountName=smiljanisp;AccountKey=1bQTlsZ2JxjnT27g5AxamdGA0gXnVhNQWN7Xm8HrBGgv+T1taOj5a+dfqSK7ap9eGhtiAN+iTmorsKDrny1HYw==;" ConcurrentWriteTest  Oak-Segment-Azure  
Apache Jackrabbit Oak 1.45-SNAPSHOT
# ConcurrentWriteTest              C     min     10%     50%     90%     max     N       mean 
Oak-Segment-Azure                  1      56      71     107     746   1816     243     247
Oak-Segment-Azure                  1      54      61      77     368   1643     379     159
```